src/classes/CON_Phone.cls

```diff
@@ -314,11 +314,25 @@ public with sharing class CON_Phone {
                 'Mobile Phone' => 'Mobile',
                 'Other Phone' => 'Other'
             };
-
+            
             this.value = val;
             this.label = lbl;
             this.apiName = api;
-            this.prefLabel = (preferredPhoneMappings.containsKey(lbl)) ? preferredPhoneMappings.get(lbl) : lbl;
+            
+            String namespace = UTIL_Namespace.getNamespace();
+            String prefix = (String.isNotBlank(namespace)) ? namespace + '__' : '';
+            List<Schema.PicklistEntry> preferredPhonePicklist = UTIL_Describe.getFieldPicklistEntries('Contact', prefix+'PreferredPhone__c');
+            List<String> preferredPhoneValues = new List<String>();
+            for (Schema.PicklistEntry eachValue: preferredPhonePicklist) {
+                preferredPhoneValues.add(eachValue.getValue());
+            }
+             
+            if (!preferredPhoneValues.contains(lbl)) {
```

Comment:
Same comment as above re: adding an explanation here or updating the comment on line 310.

Reply:
Updated it.